src/libs/actions/User.ts

```diff
@@ -623,62 +614,63 @@ function clearScreenShareRequest() {
 
 /**
  * Open an OldDot tab linking to a screen share request.
- * @param {String} accessToken Access token required to join a screen share room, generated by the backend
- * @param {String} roomName Name of the screen share room to join
+ * @param accessToken Access token required to join a screen share room, generated by the backend
+ * @param roomName Name of the screen share room to join
  */
-function joinScreenShare(accessToken, roomName) {
+function joinScreenShare(accessToken: string, roomName: string) {
     Link.openOldDotLink(`inbox?action=screenShare&accessToken=${accessToken}&name=${roomName}`);
     clearScreenShareRequest();
 }
 
 /**
  * Downloads the statement PDF for the provided period
- * @param {String} period YYYYMM format
+ * @param period YYYYMM format
  */
-function generateStatementPDF(period) {
+function generateStatementPDF(period: string) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.WALLET_STATEMENT,
+            value: {
+                isGenerating: true,
+            },
+        },
+    ];
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.WALLET_STATEMENT,
+            value: {
+                isGenerating: false,
+            },
+        },
+    ];
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.WALLET_STATEMENT,
+            value: {
+                isGenerating: false,
+            },
+        },
+    ];
     API.read(
         'GetStatementPDF',
         {period},
```

Comment:
Could you create type for this object like you did in the others?

Reply:
Bump